code/modules/virus2/disease2.dm

```diff
@@ -22,9 +22,12 @@ var/global/list/disease2_list = list()
 	disease2_list["[uniqueID]"] = src
 	..()
 
-/datum/disease2/disease/proc/new_random_effect(var/max_badness = 1, var/stage = 0)
+/datum/disease2/disease/proc/new_random_effect(var/max_badness = 1, var/stage = 0, var/datum/disease2/effect/E)
```

Comment:
you'll surely agree it's not clear that a parameter named `E` is used to exclude effects of that type.
Speaking of which, why is it an instance from which the path is obtained, rather than just being a path in the first place?